File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Python/javonet/core/receiver/Receiver.py
import sys

from javonet.core.interpreter.Interpreter import Interpreter
from javonet.core.protocol.CommandSerializer import CommandSerializer
from javonet.utils.RuntimeLogger import RuntimeLogger
from javonet.utils.connectionData.InMemoryConnectionData import InMemoryConnectionData


class Receiver:

    def __init__(self):
        RuntimeLogger.print_runtime_info()
        sys.stdout.flush()
        self.connection_data = InMemoryConnectionData()

    def SendCommand(self, message_byte_array, messageByteArrayLen):
        return bytearray(CommandSerializer().serialize(Interpreter().process(message_byte_array), self.connection_data))

    def HeartBeat(self, message_byte_array, messageByteArrayLen):
        response_byte_array = bytearray(2)
        response_byte_array[0] = message_byte_array[11]
        response_byte_array[1] = message_byte_array[12] - 2
        return response_byte_array

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Ruby/Windows/X64/core/receiver/receiver.rb
require_relative '../interpreter/interpreter'
require_relative '../../utils/runtime_logger'
require_relative '../protocol/command_serializer'

class Receiver
  
  def initialize
    RuntimeLogger.print_runtime_info
  end

  def send_command(message_array, message_array_len)
    CommandSerializer.new.serialize(Interpreter.new.process(message_array))
  end
  
  def heart_beat(message_array, message_array_len)
    [49,48]
  end
end
